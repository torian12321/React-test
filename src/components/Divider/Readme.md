[Reference](https://codepen.io/scottzirkel/pen/yNxNME?editors=0100)

#### Basic
Styled [hr](https://www.w3schools.com/tags/tag_hr.asp) component. It should be 'transparent' for SEO
```
<Divider />
```

#### Texted
```
<Divider text='Chapter 1' />
```
