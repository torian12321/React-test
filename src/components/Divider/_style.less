@import '../../styles/conf/index.less';

@themeColor_BG: @white;

.divider {
  position  : relative;
  height    : 1px;
  width     : 100%;
  margin    : 1em auto;
  border    : none;
  overflow  : visible;
  text-align: center;
  color     : @themeColor_1;
  opacity   : .8;
  background: linear-gradient(to right, transparent, @themeColor_1, transparent);

  &:before {
    @size: 8px;

    content: '';
    position  : absolute;
    left      : 50%;
    font-size : @size;
    height    : 1em;
    width     : 1em;
    margin    : -.5em;
    background: @white;
    border    : @themeColor_1 1px solid;
    box-sizing: border-box;
    .rotate(45deg);
  }
  &:after {
    content: attr(data-content);
    position   : relative;
    top        : -.8em;
    display    : inline-block;
    padding    : 0 .8em;
    line-height: 1.6em;
    background : @themeColor_BG;
  }
}