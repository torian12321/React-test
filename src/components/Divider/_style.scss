@import '../../styles/conf/index.scss';

// Vars
$divider-color    : $themeColor_1 !default;
$divider-color-sec: $themeColor_BG !default;

// Block
.divider {
  position  : relative;
  height    : 1px;
  width     : 100%;
  margin    : 1em auto;
  border    : none;
  overflow  : visible;
  text-align: center;
  color     : $divider-color;
  opacity   : .8;
  background: linear-gradient(to right, transparent, $divider-color, transparent);

  &:before {
    $size: 8px;

    content: '';
    position  : absolute;
    left      : 50%;
    font-size : $size;
    height    : 1em;
    width     : 1em;
    margin    : -.5em;
    background: $divider-color-sec;
    border    : $divider-color 1px solid;
    box-sizing: border-box;
    @include rotate(45deg);
  }
  &:after {
    content    : attr(data-content);
    position   : relative;
    top        : -.8em;
    display    : inline-block;
    padding    : 0 .8em;
    line-height: 1.6em;
    background : $divider-color-sec;
  }
}