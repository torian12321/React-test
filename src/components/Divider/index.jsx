import React      from 'react';
import PropTypes  from 'prop-types';
import classNames from 'classnames';
import './_style.scss';

class Divider extends React.PureComponent {
  render(){
    return (
      <hr
        data-content={this.props.text}
        className   ={classNames(
          'divider',
          this.props.text && 'divider--texted',
          this.props.className
        )}
      />
    );
  }
};

Divider.propTypes = {
  text     : PropTypes.string,
  className: PropTypes.string
};

module.exports = Divider;
