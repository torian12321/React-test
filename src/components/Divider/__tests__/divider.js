import React    from 'react';
import renderer from 'react-test-renderer';
import Divider  from '../index';

describe('component', () => {
  describe('<Divider />', () => {

    it('should render correctly', () => {
      expect(renderer.create(
        <Divider />
      )).toMatchSnapshot();
    });

    it('Should render a texted version', () => {
      expect(renderer.create(
        <Divider text='Chapter 1' />
      )).toMatchSnapshot();
    });

  });
});
