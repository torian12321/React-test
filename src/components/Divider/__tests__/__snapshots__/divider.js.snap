// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <Divider /> Should render a texted version 1`] = `
<hr
  className="divider divider--texted"
  data-content="Chapter 1"
/>
`;

exports[`component <Divider /> should render correctly 1`] = `
<hr
  className="divider"
  data-content={undefined}
/>
`;
